openssl/src/asn1.rs

```diff
@@ -51,6 +51,29 @@ impl fmt::Display for Asn1TimeRef {
     }
 }
 
+impl Asn1TimeRef {
+    #[cfg(any(ossl102, ossl110))]
+    pub fn as_unix(&self) -> Result<time_t, ErrorStack> {
+        Asn1Time::from_unix(0)?.diff(Some(self))
+    }
+
+    #[cfg(any(ossl102, ossl110))]
+    pub fn diff(&self, compare: Option<&Self>) -> Result<time_t, ErrorStack> {
```

Comment:
Can this return a `struct TimeDiff { days, secs }` instead? You can't generally assume days are the same length.